pydicom/tests/test_dataset.py

```diff
@@ -1834,3 +1835,24 @@ def test_show_file_meta(self):
         assert shown.startswith("(0010, 0010) Patient's Name")
 
         pydicom.config.show_file_meta = orig_show
+
+    @pytest.mark.parametrize('copy_method',
+                             [Dataset.copy, copy.copy, copy.deepcopy])
+    def test_copy(self, copy_method):
+        ds = Dataset()
+        ds.PatientName = "John^Doe"
+        ds.BeamSequence = [Dataset(), Dataset(), Dataset()]
+        ds.BeamSequence[0].Manufacturer = "Linac, co."
+        ds.BeamSequence[1].Manufacturer = "Linac and Sons, co."
+        ds.is_implicit_VR = True
+        ds.is_little_endian = True
+        ds.read_encoding = "utf-8"
+        ds_copy = copy_method(ds)
+        assert isinstance(ds_copy, Dataset)
+        assert len(ds_copy) == 2
+        assert ds_copy.PatientName == "John^Doe"
+        assert ds_copy.BeamSequence[0].Manufacturer == "Linac, co."
+        assert ds_copy.BeamSequence[1].Manufacturer == "Linac and Sons, co."
+        assert ds_copy.is_implicit_VR
+        assert ds_copy.is_little_endian
+        assert ds_copy.read_encoding == "utf-8"
```

Comment:
Can I suggest a test to confirm the shallow and deep copies?  I guess it would have to be for the `id(BeamSequence[i])` being the same or different, respectively, unless you added something like a MultiValue e.g. PixelSpacing.  With immutables like the strings used so far, the id will be the same anyway.

Reply:
Hah, actually I got this wrong - only `deepcopy` does a deep copy (sic!). I tested the wrong thing for deepcopy, obviously... The question, what shall be the correct behavior? Shall `copy` (both methods) make a shallow copy, as it is now, or shall we change the behavior?

Reply:
Just to make sure we are talking about the same thing - the only difference in 'shallow' vs 'deep' is whether references to objects are used, or a new object created and its attributes copied.  I'm seeing the correct behavior in 2.1.0.dev0 right now using the copy library imported ones, at least at top level of a dataset:

```python
>>> from copy import copy, deepcopy
>>> from pydicom.dataset import Dataset
>>> ds = Dataset()
>>> ds.PixelSpacing = [1.0, 2.0]
>>> ds_copy = copy(ds)
>>> ds_deep = deepcopy(ds)
>>> id(ds_copy.PixelSpacing) == id(ds.PixelSpacing)
True
>>> id(ds_deep.PixelSpacing) == id(ds.PixelSpacing)
False
>>>
'''

Reply:
Yes, correct. I just adapted the docstring in the last commit and adapted the test to test for shallow vs deep copy, and after your comment in the issue this is probably all what is needed.